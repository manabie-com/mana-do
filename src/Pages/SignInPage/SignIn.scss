@import url('https://fonts.googleapis.com/css2?family=Noto+Sans:wght@700&family=Poppins:wght@400;500;600&display=swap');

.SignIn {
    text-align: center;
    font-family: "Poppins", sans-serif;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 400px;
    background: white;
    border-radius: 8px;
    box-shadow: 10px 10px 15px rgba(0, 0, 0, 0.05);
}

.SingIn h1 {
    text-align: center;
    padding: 0 0 20px 0;
    border-bottom: 1px solid silver;

}

.SingIn .form {
    padding: 0 40px;
    box-sizing: border-box;
}

form .txt_field {
    position: relative;
    margin: 15px 0;
    display: inline-flex;

}

.txt_field input {
    width: 300px;
    height: 40px;
    padding: 0 5px;
    font-size: 16px;
    border: none;
    background: none;
    outline: none;
}
