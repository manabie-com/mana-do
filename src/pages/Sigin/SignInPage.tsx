import React, {useState} from 'react';

import {useHistory} from 'react-router-dom'
import Service from '../../service'

const SignInPage = () => {
    const [form, setForm] = useState({
        userId: '',
        password: ''
    });
    const history = useHistory();

    const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        const resp = await Service.signIn(form.userId, form.password)

        localStorage.setItem('token', resp)
        history.push('/todo')
    }

    const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.persist()
        setForm(prev=>({
            ...prev,
            [e.target.name]: e.target.value
        }))
    }

    return (
        <div style={{marginTop: '3rem', textAlign: 'left'}}>
            <form onSubmit={signIn}>
                <label htmlFor="user_id">
                    User id
                    <input
                        id="user_id"
                        name="userId"
                        value={form.userId}
                        style={{marginTop: 12}}
                        onChange={onChangeField}
                    />
                </label>
                <br/>
                <label htmlFor="password" >
                    Password
                    <input
                        id="password"
                        name="password"
                        type="password"
                        style={{marginTop: 12}}
                        value={form.password}
                        onChange={onChangeField}
                    />
                </label>
                <br />
                <button type="submit" style={{marginTop: 12}}>
                    Sign in
                </button>
            </form>
        </div>
    );
};

export default SignInPage;