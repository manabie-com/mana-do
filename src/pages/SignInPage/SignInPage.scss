@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.sign-in {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  h1 {
    color: $primiry-color;
  }

  &__container {
    padding: 8px 24px 28px 24px !important;
  }
  &__form {
    flex: 3;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    @include max-respond-to(md) {
      flex: 1;
    }
  }
  &__img {
    width: 50%;
    flex: 2;
    @include max-respond-to(md) {
      display: none;
      flex: 0;
    }
  }
  &__input {
    width: 300px !important;
  }
  &__action {
    display: flex;
    justify-content: center;
    align-items: center;
  }
  &__welcome {
    margin-bottom: 32px;
  }
}
