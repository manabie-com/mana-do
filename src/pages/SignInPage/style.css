.sign-in-page {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url("/images/SignIn/background.webp");
  background-repeat: no-repeat;
  background-size: cover;
}

.sign-in-page .form-login {
  width: 30%;
}

.form-login .header {
  margin-bottom: 20px;
}

.form-control {
  margin-bottom: 20px;
}

.form-control label {
  font-weight: 500;
  font-size: 14px;
}

.form-control div {
  margin-top: 5px;
}
