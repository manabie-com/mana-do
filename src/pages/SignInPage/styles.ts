import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;

  .Login__inner {
    width: 350px;
    margin: 24px auto;
    box-shadow: 0 10px 25px rgba(0, 0, 0, 0.05), 0 20px 48px rgba(0, 0, 0, 0.05), 0 1px 4px rgba(0, 0, 0, 0.1);
    background: var(--listbg);
    padding: 20px;
    text-align: left;
    color: var(--font-color);

    .Login__row {
      margin: 0 0 24px;
    }

    input {
      width: 100%;
      padding: 18px;
      background: transparent;
      border-radius: 4px;
      border: 1px solid var(--darkgrayishblue);
      flex: 1 0 auto;
      color: var(--font-color);
      font-size: 14px;
      font-family: inherit;

      :focus,
      :active {
        outline: none;
      }
    }

    button {
      width: 100%;
      min-height: 32px;
      min-width: 80px;
      padding: 4px 8px;
      cursor: pointer;
      display: inline-block;
      outline: 0;
      border: none;
      background: #e0e1e2 none;
      color: rgba(0, 0, 0, 0.6);
      margin: 0 4px 0 0;
      text-transform: none;
      text-shadow: none;
      font-weight: 700;
      line-height: 32px;
      font-style: normal;
      text-align: center;
      text-decoration: none;
      border-radius: 4px;
      background: var(--brightblue);
      box-shadow: 0 0 0 1px rgba(34, 36, 38, 0.15) inset;
      color: white;
    }
  }
`;
