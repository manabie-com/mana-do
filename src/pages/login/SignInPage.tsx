import React, { useContext, useState } from 'react';
import { useHistory, Link } from 'react-router-dom'
import Service from '../../service'
import AppButton from '../../commons/AppButton';
import AppInput from '../../commons/AppInput';
import { actLogin } from '../../store/actions';
import { AppContext } from '../../store/context';
import AppFooter from '../../components/AppFooter';
import AppAlert from '../../commons/AppAlert';
import { useNotAuthenticated } from '../../hooks/useNotAuthenticated';

const SignInPage = () => {
    useNotAuthenticated();
    const [form, setForm] = useState({
        userId: '',
        password: ''
    });
    const { dispatch } = useContext(AppContext);
    const history = useHistory();
    const [loading, setLoading] = useState(false);

    const [alertMsg, setAlert] = useState({
        visible: false,
        title: '',
        content: '',
        type: ''
    } as {[x:string]:any});

    const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        setLoading(true)
        try {
            const response = await Service.signIn(form.userId, form.password);
            if (response.status === 200) {
                dispatch(actLogin(response.data));
                setLoading(false)
                history.push('/todo');
            } 
        } catch (error) {
            setAlert({
                visible:true,
                type: 'error',
                title: 'Error',
                content: 'Authentication failed',   //THIENNGUYEN: replace error.message by a friendly message
            })
            setLoading(false)
        }
        
    }

    const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.persist()
        setForm(prev => ({
            ...prev,
            [e.target.name]: e.target.value
        }))
    }

    

    return (
        <main className="login">
            <div className="ToDo__container">
                <div style={{margin: "auto 32px"}}>
                    <div style={{ textAlign: "center" }}>
                        <img alt="React logo" src='logo192.png' style={{ width: 100 }}></img>
                        <h4>Please sign in to continue</h4>
                        {
                            alertMsg.visible && <AppAlert alertType={alertMsg.type} content={alertMsg.content}></AppAlert>
                        }
                    </div>
                    <form onSubmit={signIn}>
                            <AppInput
                                id="user_id"
                                name="userId"
                                value={form.userId}
                                style={{ marginTop: 12 }}
                                onChange={onChangeField}
                                placeholder="Please enter your username..."
                                label="Username"
                                required
                            />
                        <br />
                            <AppInput
                                id="password"
                                name="password"
                                type="password"
                                placeholder="Please enter your password"
                                style={{ marginTop: 12 }}
                                value={form.password}
                                onChange={onChangeField}
                                required
                                label="Password"
                            ></AppInput>
                        <br />
                        <AppButton isLoading={loading} style={{ minWidth: "100%" }} btnType="primary" type="submit">Sign in</AppButton>
                        <div style={{textAlign: "center"}}>
                        <span>Not a member? </span><Link to="/register" >Register an acoount</Link>
                        </div>
                        <AppFooter textColor="theme" />
                    </form>
                </div>
            </div>
        </main>
    );
};

export default SignInPage;