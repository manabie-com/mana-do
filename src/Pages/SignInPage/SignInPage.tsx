import React, { useState } from 'react';

import { useHistory } from 'react-router-dom'
import Service from '../../service';
import './SignIn.scss';

const SignInPage = () => {
    const [form, setForm] = useState({
        userId: '',
        password: ''
    });
    const history = useHistory();

    const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        const resp = await Service.signIn(form.userId, form.password)

        localStorage.setItem('token', resp)
        history.push('/todo')
    }

    const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.persist()
        setForm(prev => ({
            ...prev,
            [e.target.name]: e.target.value
        }))
    }

    return (
        <>
            <div className="SignIn">
                <h1>Sign In</h1>
                <form onSubmit={signIn} >
                    <div className="txt_field">
                        <input
                            id="user_id"
                            name="userId"
                            value={form.userId}
                            // style={{ marginTop: 12 }}
                            onChange={onChangeField}
                            placeholder={"User Id"}
                        />
                        {/* <label htmlFor="user_id">
                            User id
                        </label> */}
                    </div>
                    <div className="txt_field">
                        {/* <label htmlFor="password">
                            Password </label> */}

                        <input
                            id="password"
                            name="password"
                            type="password"
                            // style={{ marginTop: 12 }}
                            value={form.password}
                            onChange={onChangeField}
                            placeholder={"Password"}
                        />
                    </div>

                    <br />
                    <button
                        type="submit"
                    // style={{ marginTop: 12 }}
                    >
                        Sign in
                    </button>

                </form>

            </div>
        </>
    );
};

export default SignInPage;