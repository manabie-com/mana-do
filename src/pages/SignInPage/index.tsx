import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { updateTokenLocalStorage, getTokenLocalStorage } from "utils/storage";
import Service from "service";

import { Button, Input, Paper } from "components/commons";

import "./style.css";

const SignInPage = () => {
  const [form, setForm] = useState({
    userId: "",
    password: "",
  });
  const history = useHistory();
  const token = getTokenLocalStorage();

  if (token) {
    history.push("/todo");
  }

  const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const resp = await Service.signIn(form.userId, form.password);
    updateTokenLocalStorage(resp);
    history.push("/todo");
  };

  const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.persist();
    setForm((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <div className="sign-in-page">
      <Paper className="form-login p30">
        <div className="text-center header">
          <img src="images/SignIn/logo.svg" alt="logo" />
        </div>

        <form onSubmit={signIn}>
          <div className="form-control">
            <label>Username</label>
            <Input
              id="user_id"
              name="userId"
              value={form.userId}
              onChange={onChangeField}
            />
          </div>

          <div className="form-control">
            <label>Password</label>
            <Input
              id="password"
              name="password"
              type="password"
              value={form.password}
              onChange={onChangeField}
            />
          </div>

          <div className="text-right">
            <Button type="submit">Sign in</Button>
          </div>
        </form>
      </Paper>
    </div>
  );
};

export default SignInPage;
