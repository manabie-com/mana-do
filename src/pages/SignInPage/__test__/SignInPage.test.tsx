import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";

import SignInPage from "pages/SignInPage";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

let container: HTMLElement;

beforeEach(() => {
  container = render(<SignInPage />).container;
});

afterEach(cleanup);

test("SignInPage: render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SignInPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("SignInPage: change input username", () => {
  const usernameEl = container.querySelector("#user_id") as HTMLInputElement;
  fireEvent.change(usernameEl, { target: { value: "test" } });
  expect(usernameEl.value).toEqual("test");
});

test("SignInPage: change input password", () => {
  const passwordEl = container.querySelector("#user_id") as HTMLInputElement;
  fireEvent.change(passwordEl, { target: { value: "123456" } });
  expect(passwordEl.value).toEqual("123456");
});

test("SignInPage: match snapshot", () => {
  const tree = renderer.create(<SignInPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
