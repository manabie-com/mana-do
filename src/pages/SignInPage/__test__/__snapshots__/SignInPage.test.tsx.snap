// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage: match snapshot 1`] = `
<div
  className="sign-in-page"
>
  <div
    className="paper-custom form-login p30 paper-custom-elevation-1"
  >
    <div
      className="text-center header"
    >
      <img
        alt="logo"
        src="images/SignIn/logo.svg"
      />
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-control"
      >
        <label>
          Username
        </label>
        <div
          className="root-input"
        >
          <input
            className="input-custom input-size-md"
            id="user_id"
            name="userId"
            onChange={[Function]}
            value=""
          />
        </div>
      </div>
      <div
        className="form-control"
      >
        <label>
          Password
        </label>
        <div
          className="root-input"
        >
          <input
            className="input-custom input-size-md"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="text-right"
      >
        <button
          className="btn-custom btn-custom-primary"
          disabled={false}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
</div>
`;
